src/common/libflux/reactor_private.h

```diff
@@ -0,0 +1,67 @@
+/************************************************************\
+ * Copyright 2021 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#ifndef _FLUX_CORE_REACTOR_PRIVATE_H
+#define _FLUX_CORE_REACTOR_PRIVATE_H
+
+#include "src/common/libev/ev.h"
+#include "reactor.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct flux_reactor {
+    struct ev_loop *loop;
+    int usecount;
+    unsigned int errflag:1;
+};
+
+struct flux_watcher {
+    flux_reactor_t *r;
+    flux_watcher_f fn;
+    void *arg;
+    struct flux_watcher_ops *ops;
+    void *data;
+};
```

Comment:
It's a bit unfortunate that we need to expose these things outside of reactor.c, given that we theoretically are providing an API to implement custom watchers.  I guess it's because ev_zmq was implemented as a pure libev watcher, and libev isn't exposed directly.   We _might_ be able to reimplement this in terms of the exported functions, but I'm thinking that could be hairy and time consuming, and this works.

I guess it's for the greater good, and it's still private to flux-core :-)

Reply:
i didn't think about the idea of implementing via exposed functions.  But yeah, the fact `libev` wasn't public did create problems.

Reply:
Not worth the effort at this time IMHO.